### PR TITLE
Consolidate nodeagent_service proto

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

File: security/proto/nodeagent_service.proto [hunk 1]
```diff
@@ -21,6 +21,12 @@ option go_package="istio_v1_auth";
 
 service NodeAgentService {
 	rpc Check(CheckRequest) returns (CheckResponse);
```
Comment (myidpt) [inline security/proto/nodeagent_service.proto]: Can you also add comment for Check?

File: security/proto/nodeagent_service.proto [hunk 2]
```diff
@@ -21,6 +21,12 @@ option go_package="istio_v1_auth";
 
 service NodeAgentService {
```
Comment (myidpt) [inline security/proto/nodeagent_service.proto]: Can you add a short comment on the purpose this service?

**Comment by ZackButcher** [comment]
Drive by comment since I saw this due to global reviewers: why do these definitions live in istio/istio and not istio/api? I get that the node agent stuff probably shouldn't since the user doesn't interact with it, but it feels like the security service definition should be in istio/api.

**Comment by wattli** [comment]
@ZackButcher , we had some discussion on this, the idea is to keep security internal proto in this repo, please refer here for more info: https://github.com/istio/istio/pull/2809

File: security/proto/nodeagent_service.proto (hunk 1 above)
Comment (wattli) [reply security/proto/nodeagent_service.proto]: done, thanks.

File: security/proto/nodeagent_service.proto (hunk 2 above)
Comment (wattli) [reply security/proto/nodeagent_service.proto]: done, thanks

**Comment by saumoh** [comment]
@wattli The change it self looks fine. I am just wondering if tying in the interface between flex-volume and the node agent with other node agent services is the right thing to do?
Flex volume is very k8s specific and the rest of the node agent is still very platform agnostic. wdyt?

**Comment by wattli** [comment]
The intention is to avoid more services, currently we have two components, ca and nodeagent, and it would make the structure more readable to have ca_service and nodeagent_service.

**Pull request merged**